realm/realm-library/src/main/cpp/object-store/src/object_store.hpp

```diff
@@ -75,6 +76,36 @@ namespace realm {
         static std::string table_name_for_object_type(StringData class_name);
         static StringData object_type_for_table_name(StringData table_name);
 
+        // Those are needed by java for the async query
```

Comment:
@bdash Would those look OK? Or simply expose the `SharedGroup` pointer and keep those functions in JNI to make the Object Store API cleaner?